pl_bolts/models/gans/srgan/components.py

```diff
@@ -0,0 +1,150 @@
+            nn.Conv2d(feature_maps, feature_maps, kernel_size=3, padding=1),
+            nn.BatchNorm2d(feature_maps),
+        ]
+        self.res_blocks = nn.Sequential(*res_blocks)
+
+        # PixelShuffle blocks (k3n256s1)
+        ps_blocks = []
+        for _ in range(num_ps_blocks):
+            ps_blocks += [
+                nn.Conv2d(feature_maps, 4 * feature_maps, kernel_size=3, padding=1),
+                nn.PixelShuffle(2),
+                nn.PReLU(),
+            ]
+        self.ps_blocks = nn.Sequential(*ps_blocks)
+
+        # Output block: k9n3s1
+        self.output_block = nn.Sequential(
+            nn.Conv2d(feature_maps, image_channels, kernel_size=9, padding=4),
+            nn.Tanh(),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x_res = self.input_block(x)
+        x = x_res + self.res_blocks(x_res)
+        x = self.ps_blocks(x)
+        x = self.output_block(x)
+        return x
+
+
+class SRGANDiscriminator(nn.Module):
+    def __init__(self, image_channels: int, feature_maps: int = 64) -> None:
+        super().__init__()
+        # k3n64s1 --> k3n64s2 --> k3n128s1 --> k3n128s2 --> k3n256s1 --> k3n256s2 --> k3n512s1 --> k3n512s2 --> MLP
+
+        self.conv_blocks = nn.Sequential(
+            # k3n64s1 --> k3n64s2
+            self._make_double_conv_block(image_channels, feature_maps, first_batch_norm=False),
+            # k3n128s1 --> k3n128s2
+            self._make_double_conv_block(feature_maps, feature_maps * 2),
+            # k3n256s1 --> k3n256s2
+            self._make_double_conv_block(feature_maps * 2, feature_maps * 4),
+            # k3n512s1 --> k3n512s2
+            self._make_double_conv_block(feature_maps * 4, feature_maps * 8),
+        )
+
+        self.mlp = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1),
+            nn.Conv2d(feature_maps * 8, feature_maps * 16, kernel_size=1),
+            nn.LeakyReLU(0.2, inplace=True),
+            nn.Conv2d(feature_maps * 16, 1, kernel_size=1),
+            nn.Flatten(),
+        )
+
+    def _make_double_conv_block(
+        self, in_channels: int, out_channels: int, first_batch_norm: bool = True
+    ) -> nn.Sequential:
+        return nn.Sequential(
+            self._make_conv_block(in_channels, out_channels, batch_norm=first_batch_norm),
+            self._make_conv_block(out_channels, out_channels, stride=2),
+        )
+
+    @staticmethod
+    def _make_conv_block(
+        in_channels: int, out_channels: int, stride: int = 1, batch_norm: bool = True
+    ) -> nn.Sequential:
+        return nn.Sequential(
+            nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=stride, padding=1),
+            nn.BatchNorm2d(out_channels) if batch_norm else nn.Identity(),
+            nn.LeakyReLU(0.2, inplace=True),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.conv_blocks(x)
+        x = self.mlp(x)
+        return x
+
+
+class VGG19FeatureExtractor(nn.Module):
```

Comment:
I think FeatureExtractor using this method is also common.
I think some part of this code matches with #475